pkg/internal/server/grpc/converters.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package grpc
+
+import (
+	"context"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/kubernetes-csi/external-snapshot-metadata/pkg/api"
+)
+
+func newCSIGetMetadataDeltaRequest(ctx context.Context, baseSnapHandle, targetSnapHandle string, req *api.GetMetadataDeltaRequest) *csi.GetMetadataDeltaRequest {
+	return &csi.GetMetadataDeltaRequest{
+		BaseSnapshotId:   baseSnapHandle,
+		TargetSnapshotId: targetSnapHandle,
+		StartingOffset:   req.StartingOffset,
+		MaxResults:       req.MaxResults,
+		//TODO: set Secrets field
+	}
+}
```

Comment:
I think this function signature should be:
```
func newCSIGetMetadataDeltaRequest(ctx context.Context, req *api.GetMetadataDeltaRequest) (*csi.GetMetadataDeltaRequest, error) { ... }
```
i.e. it does the conversion of snapshot handles that you added to the validator.